### DISCUSSION TITLE
RSuiteJS - ERESOLVE unable to resolve dependency tree

### DISCUSSION THREAD

**Discussion by alexshenker** [rsuite/rsuite#1866] [Q&A] [answered]
I've seen workarounds for this that suggest running the installation with '**--legacy-peer-deps**' and/or **--force**. However, I'd really
rather understand what the underlying issue is and if a fix is coming. I am interested in replacing Material-Ui with rsuite, but I'd rather not do so with force or legacy peer.
Thoughts?

**Work station:**
- macbook pro 2021, Big Sur
- React ^17.0.2

**Error:**
alex@alex-MacBook-Pro client % **npm i rsuite --save**                   
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   peer react@"*" from @testing-library/react@11.2.7
npm ERR!   node_modules/@testing-library/react
npm ERR!     @testing-library/react@"^11.2.7" from the root project
npm ERR!   peer react@"^0.14.0 || ^15.0.0 || ^16.0.0 || ^17.0.0" from mini-create-react-context@0.4.1
npm ERR!   node_modules/mini-create-react-context
npm ERR!     mini-create-react-context@"^0.4.0" from react-router@5.2.1
npm ERR!     node_modules/react-router
npm ERR!       react-router@"5.2.1" from react-router-dom@5.2.1
npm ERR!       node_modules/react-router-dom
npm ERR!         react-router-dom@"^5.2.1" from the root project
npm ERR!   6 more (react-dom, react-router, react-router-dom, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^15.3.0 || ^16.0.0-alpha" from react-virtualized@9.22.3
npm ERR! node_modules/rsuite/node_modules/react-virtualized
npm ERR!   react-virtualized@"^9.21.0" from rsuite@4.10.2
npm ERR!   node_modules/rsuite
npm ERR!     rsuite@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/alex/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/alex/.npm/_logs/2021-08-31T16_11_33_446Z-debug.log
alex@alex-MacBook-Pro client %

Thank you

**Comment by SevenOutman** [comment, marked as answer]
The log shows that `react-virtualized` - a dependency of rsuite v4 - doesn’t have a peer dependency of React 17, and there’s not yet a update release adding support for it. See related issue https://github.com/bvaughn/react-virtualized/issues/1655 .

Since React 17 doesn’t come with breaking changes, I believe it’s safe to just use —legacy-peer-does or —force flags before we can upgrade this dependency.
